src/libcore/cell.rs

```diff
@@ -1131,6 +1187,61 @@ impl<'a, T: ?Sized + fmt::Display> fmt::Display for RefMut<'a, T> {
     }
 }
 
+#[stable(feature = "std_guard_impls_ext", since = "1.23.0")]
+impl<'a, T: ?Sized + Hash> Hash for RefMut<'a, T> {
+    #[inline]
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.value.hash(state);
+    }
+}
+
+#[stable(feature = "std_guard_impls_ext", since = "1.23.0")]
+impl<'a, T: ?Sized + PartialEq> PartialEq for RefMut<'a, T> {
+    #[inline]
+    fn eq(&self, other: &RefMut<T>) -> bool {
+        self.value == other.value
+    }
+}
+
+#[stable(feature = "std_guard_impls_ext", since = "1.23.0")]
+impl<'a, T: ?Sized + Eq> Eq for RefMut<'a, T> { }
+
+#[stable(feature = "std_guard_impls_ext", since = "1.23.0")]
+impl<'a, T: ?Sized + Ord> Ord for RefMut<'a, T> {
+    #[inline]
+    fn cmp(&self, other: &RefMut<T>) -> Ordering {
+        self.value.cmp(&other.value)
+    }
+}
+
+#[stable(feature = "std_guard_impls_ext", since = "1.23.0")]
```

Comment:
Unless you expect this PR will stay open for 20 days, the `since` field should still be 1.22.0.

Reply:
Done, thanks!